File: 1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
    dict := make(map[int]int)
    
    for index, value := range nums {
        if diffIndex, ok := dict[target - value]; ok {
            return []int{diffIndex, index}
        } else {
            dict[value] = index;
        }
    }
    
    return nil;
}

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const memo = new Map();
    
    for (let i = 0; i < nums.length; i++) {
        if (memo.get(target - nums[i]) !== undefined) 
            return [memo.get(target - nums[i]), i];
        memo.set(nums[i], i);
    }
};